### ISSUE TITLE
Don't optimize import React = require("react") away in case of TSX

### ISSUE THREAD

**Issue by rogierschouten** [Microsoft/TypeScript#6204]
The file below will compile  (--target es5, --module commonjs, --jsx "preserve")  but not run, because the require statement is optimized away. 

<EDIT>
 In the subsequent JSX translation, the `<div></div>` elements are translated to `React.createElement()` calls which are then no longer possible.
</EDIT>

```
import React = require("react");

export function render(): JSX.Element {
   return (<div></div>);
}
```
My current workaround is this:
```
/* tslint:disable:no-unused-expressions */
import React = require("react"); React
/* tslint:enable:no-unused-expressions */

export function render(): JSX.Element {
   return (<div></div>);
}
```
but that is rather cumbersome due to the tslint errors you will get.

Therefore, maybe not optimize React require away in case of .tsx?

**Comment by RyanCavanaugh** [comment]
We already preserve the import when using `--jsx react`. We need a more general solution here since we don't really know what the import name will be if you're using an external JSX -> JS transformer.

Out of curiosity, why are you using `---jsx preserve` if you're targeting React anyway?

**Comment by rogierschouten** [comment]
I'm using "preserve" because I'm rewriting a 150kloc code base that still has many JSX files, so I have to use Babel anyway while not all JSX is gone. 

Also, we're starting to feel the pain of tools linked too closely - for instance, upgrading TypeScript to a new version requires us to wait for tslint, typedoc and ts-reflect and sometimes their gulp plugins to be upgraded to the new version before we can make the move. Imagine also having to upgrade React because the jsx transform has changed in either React or TypeScript

**Comment by mhegazy** [comment]
Duplicate of #5865. should be fixed in https://github.com/Microsoft/TypeScript/pull/6290.
